File: codigos_c++/taktaktak.cpp
#include <bits/stdc++.h>

using namespace std;

string tumba;

int main(){
    //primeiro, leitura
    string senha;
    cin >> tumba;
    int tam = 1;

    senha.push_back(tumba[0]);
    for(size_t i = 0; i < tumba.length()*2; i++){

        if(senha[i%tam] != tumba[i%tumba.length()]){

            for(int j = senha.size(); j <= i; j++){
                if(i-tam >= tumba.length()/2){
                    cout << "-1\n";
                    return 0;
                }
                senha.push_back(tumba[j]);
                tam++;
                if(senha[i%tam] == tumba[i]) break;
            }
        }
    }

    if(senha.size() > tumba.length()/2){
        cout << "-1\n";
        return 0;
    }

    int maior_div = 0;
    for(size_t i = 1; i <= tumba.length()/2; i++){
        if(i%senha.size() == 0 && tumba.length()%i == 0){
            maior_div = (int) i;
        }
    }

    if(maior_div == 0){
        cout << "-1\n";
        return 0;
    }

    for(int n = 0; n < maior_div; n++){
        cout << senha[n%senha.size()];
    }
    cout << endl;

    return 0;
}

File: codigos_python/taktaktak.py
try:
    while True:
        tumba = input()

        tam = 1
        n = len(tumba)

        for i in range(n):
            while tumba[i%tam] != tumba[i] and tam < n:
                tam += 1



# senha = []

# tumba = input()

# tam = 1

# senha.append(tumba[0])

# for i in range(len(tumba)):
#     if senha[i%tam] != tumba[i]:
#         for j in range(len(senha), i+1):
#             senha.append(tumba[j])
#             tam = tam + 1
#             if senha[i%tam] == tumba[i]: break

        # if len(senha) > len(tumba)/2:
        #     print(-1)
        #     exit(0)

        # maior_div = 0

        # for i in range(1, len(tumba)//2 + 1):
        #     if i%len(senha) == 0 and len(tumba)%i == 0:
        #         maior_div = i

        # if maior_div == 0:
        #     print(-1)
        #     exit(0)

        # maior_div = maior_div // len(senha)

        # print(''.join(senha)*maior_div)

except EOFError:
    pass